turbopack/crates/turbopack/src/lib.rs

```diff
@@ -683,7 +636,12 @@ impl AssetContext for ModuleAssetContext {
             request,
             resolve_options,
         );
-        let mut result = self.process_resolve_result(result.resolve().await?, reference_type);
+
+        let mut result = self.process_resolve_result(
+            result.resolve().await?,
+            reference_type,
+            request.request_pattern().await?.has_dynamic_parts(),
```

Comment:
I think there need to be an option to control that behavior:

For NFT we want to ignore unknown modules everywhere.
For Turbopack we want to make unknown modules warnings in dynamic expressions only.